test/unit/model/compile/QtiItemAssetCompilerTest.php

```diff
@@ -0,0 +1,226 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\test\unit\model\compile;
+
+use oat\generis\test\TestCase;
+use oat\oatbox\filesystem\Directory;
+use oat\oatbox\filesystem\File;
+use oat\oatbox\log\LoggerService;
+use oat\tao\model\media\MediaAsset;
+use oat\tao\model\media\MediaBrowser;
+use oat\taoItems\model\media\ItemMediaResolver;
+use oat\taoQtiItem\model\compile\QtiAssetCompiler\QtiItemAssetCompiler;
+use oat\taoQtiItem\model\compile\QtiItemCompilerAssetBlacklist;
+use oat\taoQtiItem\model\pack\QtiAssetPacker\PackedAsset;
+use oat\taoQtiItem\model\qti\Img;
+use oat\taoQtiItem\model\qti\Item;
+use oat\taoQtiItem\model\qti\XInclude;
+use oat\taoQtiItem\test\unit\model\compile\mock\ElementMock;
+use Psr\Log\NullLogger;
+
+class QtiItemAssetCompilerTest extends TestCase
+{
+    /** @var QtiItemAssetCompiler */
+    private $subject;
+
+    /** @var QtiItemCompilerAssetBlacklist */
+    private $blackListService;
+
+    /** @var ItemMediaResolver */
+    private $resolver;
+
+    /** @var Item */
+    private $item;
+
+    /** @var Directory */
+    private $directory;
+
+    public function setUp(): void
+    {
+        $this->subject = new QtiItemAssetCompiler();
+
+        $this->blackListService = $this->createMock(QtiItemCompilerAssetBlacklist::class);
+        $this->subject->setServiceLocator($this->getServiceLocatorMock([
+            QtiItemCompilerAssetBlacklist::SERVICE_ID => $this->blackListService,
+            LoggerService::SERVICE_ID => new NullLogger()
+        ]));
+
+        $this->resolver = $this->createMock(ItemMediaResolver::class);
+        $this->item = $this->createMock(Item::class);
+        $this->directory = $this->createMock(Directory::class);
+    }
+
+    public function testExtractAndCopyAssetFiles()
+    {
+        $this->item
+            ->method('getComposingElements')
+            ->willReturn([
+                (new ElementMock())->setComposingElements([
+                    $this->createConfiguredMock(XInclude::class, ['attr' => 'stimulus-href'])
+                ]),
+                (new ElementMock())->setComposingElements([
+                    $this->createConfiguredMock(Img::class, ['attr' => 'image-src'])
+                ])
+            ]);
+
+        $this->resolver->expects($this->exactly(2))
+            ->method('resolve')
+            ->willReturnOnConsecutiveCalls(
+                new MediaAsset(
+                    $this->createConfiguredMock(MediaBrowser::class, ['getFileInfo' => ['link' => 'stimulus-link']]),
+                    'stimulus-fixture'
+                ),
+                new MediaAsset(
+                    $this->createConfiguredMock(MediaBrowser::class, ['getFileInfo' => ['link' => 'image-link']]),
+                    'image-fixture'
+                ),
```

Comment:
Extra comma that breaks the test
```suggestion
                )
```